libyul/optimiser/Semantics.cpp

```diff
@@ -102,33 +102,13 @@ map<YulString, SideEffects> SideEffectsPropagator::sideEffects(
 	// is actually a bit different from "not movable".
 
 	map<YulString, SideEffects> ret;
-	for (auto const& function: _directCallGraph.functionsWithLoops)
+	for (auto const& function: _directCallGraph.functionsWithLoops + _directCallGraph.recursiveFunctions())
```

Comment:
Can you pull this out into another PR?

Reply:
https://github.com/ethereum/solidity/pull/9427